lisp/ess-julia.el

```diff
@@ -364,6 +364,52 @@ It makes underscores and dots word constituent chars.")
 
 (defvar ess-julia-mode-syntax-table (copy-syntax-table julia-mode-syntax-table))
 
+(defvar ess-julia-function-pattern
+  ;; (rx (or "function"
+  ;;         "macro"
+  ;;         (seq (0+ nonl) "(" (1+ nonl) ")" (0+ whitespace) "=")))
+  "function\\|macro\\|.*(.+)[[:space:]]*="
```

Comment:
The regex for parameters should allow for empty parens, eg. `(.*)` to support `f() = 1`. Also, something like
```
f(x
  ) =
      x +
      1
```
is legal syntax, so something like `([^)]*)` might be better.